symfony/plugins/orangehrmRecruitmentPlugin/Api/VacancyAPI.php

```diff
@@ -0,0 +1,387 @@
+    {
+        return new ParamRuleCollection(new ParamRule(CommonParams::PARAMETER_ID));
+    }
+
+    /**
+     * @inheritDoc
+     * @throws DaoException
+     */
+    public function getAll(): EndpointCollectionResult
+    {
+        $vacancyParamHolder = new VacancySearchFilterParams();
+        $this->setSortingAndPaginationParams($vacancyParamHolder);
+
+        $vacancyParamHolder->setJobTitleId(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_JOB_TITLE_ID
+            )
+        );
+        $vacancyParamHolder->setEmpNumber(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_HIRING_MANAGER_ID
+            )
+        );
+        $vacancyParamHolder->setVacancyId(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_VACANCY_ID
+            )
+        );
+        $vacancyParamHolder->setStatus(
+            $this->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_STATUS
+            )
+        );
+//        $status = $this->getRequestParams()->getIntOrNull(
+//            RequestParams::PARAM_TYPE_QUERY,
+////            self::FILTER_STATUS
+//        );
+//        if (!is_null($status)) {
+//            $vacancyParamHolder->setStatus(1);
+//        } else {
+//            $vacancyParamHolder->setStatus(0);
+//        }
+
+        $vacancies = $this->getVacancyService()->getAllVacancies($vacancyParamHolder);
+        $count = $this->getVacancyService()->getVacancyDao()->searchVacanciesCount($vacancyParamHolder);
+//        var_dump($count);
+        return new EndpointCollectionResult(
+            VacancyModel::class,
+            $vacancies,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+//            $this->getValidationDecorator()->notRequiredParamRule(
+//                new ParamRule(self::FILTER_VACANCY_ID)
+//            ),
+//            $this->getValidationDecorator()->notRequiredParamRule(
+//                new ParamRule(self::FILTER_HIRING_MANAGER_ID)
+//            ),
+//            $this->getValidationDecorator()->notRequiredParamRule(
+//                new ParamRule(self::FILTER_JOB_TITLE_ID)
+//            ),
+//            $this->getValidationDecorator()->notRequiredParamRule(
+//                new ParamRule(self::FILTER_STATUS)
+//            ),
+            new ParamRule(self::FILTER_VACANCY_ID),
+            new ParamRule(self::FILTER_HIRING_MANAGER_ID),
+            new ParamRule(self::FILTER_JOB_TITLE_ID),
+            new ParamRule(self::FILTER_STATUS),
```

Comment:
need to define rules. e.g. `new ParamRule(self::FILTER_VACANCY_ID, new Rule(Rules::POSITIVE)),`